volumes.cpp

```diff
@@ -839,16 +838,17 @@ class RecVolume : public SurfaceVolume {
   Vector3d base_center;
   Transform transform;
   double length, radius1, radius2;
+  bool facet;
 
 public:
-  RecVolume( const Vector3d& center_p, const Vector3d& axis, const Vector3d& v1, const Vector3d& v2 ) :
+  RecVolume( const Vector3d& center_p, const Vector3d& axis, const Vector3d& v1, const Vector3d& v2, bool facet_p ) :
```

Comment:
It looks like the `facet` variable is only used to set the length of the cylinder when making a facet instead of a full macrobody.  Maybe you can just initialize the length in this constructor based on that variable and won't need a member variable `facet`  (same for `RccVolume`)